packages/flutter_test/lib/src/widget_tester.dart

```diff
@@ -463,6 +464,88 @@ class WidgetTester extends WidgetController implements HitTestDispatcher, Ticker
     });
   }
 
+  @override
+  Future<void> handlePointerEventPack(List<PointerEventPack> records) {
+    assert(records != null);
+    assert(records.isNotEmpty);
+    // hitTestHistory is an equivalence of _hitTests in [GestureBinding]
+    final Map<int, HitTestResult> hitTestHistory = <int, HitTestResult>{};
+    return TestAsyncUtils.guard<void>(() async {
+      final DateTime startTime = binding.clock.now();
+      for(final PointerEventPack pack in records) {
+        final Duration timeDiff = pack.timeStamp - binding.clock.now().difference(startTime);
+        if (timeDiff.isNegative) {
+          // Flush all past events
+          // maybe trigger a warning
+          for (final PointerEvent event in pack.events) {
+            _handlePointerEvent(event, hitTestHistory);
+          }
+        }
+        else {
+          await binding.executeLater(timeDiff, () async {
+            for (final PointerEvent event in pack.events) {
+              _handlePointerEvent(event, hitTestHistory);
+            }
+          });
+          // TODO(CareF): delete the if sentence when
+          // https://github.com/flutter/flutter/issues/60739 is fixed
+          if (binding is LiveTestWidgetsFlutterBinding &&
+              (binding as LiveTestWidgetsFlutterBinding).framePolicy ==
+              LiveTestWidgetsFlutterBindingFramePolicy.fullyLive) {
+            continue;
+          }
+          await binding.pump();
+        }
+      }
+      // This make sure that a gesture is completed, with no more pointers
+      // active.
+      assert(hitTestHistory.isEmpty);
+    });
+  }
+
+  // This is a parallele implementation of [GestureBinding._handlePointerEvent]
+  // to make compatible with test bindings.
+  void _handlePointerEvent(
```

Comment:
As we discussed today, we probably want to simulate at the `_handlePointerDataPacket` level instead of the `_handlePointerEvent`?

Reply:
I still feel it's better to leave the testing API `PointerEvent` and do the conversion when accepting json input, as #60741 (I'm keeping this PR up-to-date with our discussion), because this API is also useful as accepting a controlled sequence of input for testing, see my test case, and `PointerEvent` is easy to write and `PointerData` is raw without logical pixel conversion, history dependent to decide type, etc. Also, for release mode binding, `PointerData`->`PointerEvent` conversion happens before anything goes into loops for hittest or widget reaction.

Reply:
Sounds good.